src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,882 @@
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OpenSubMenu.", this));
+#endif // MFSI_DEBUG
+
+			EnsurePopupAndPresenter();
+			EnsureCloseExistingSubItems();
+
+			MenuFlyoutPresenter parentMenuFlyoutPresenter;
+			parentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+
+			if (parentMenuFlyoutPresenter != null)
+			{
+				IMenu owningMenu;
+				owningMenu = (parentMenuFlyoutPresenter as IMenuPresenter).OwningMenu;
+				(m_tpPresenter as IMenuPresenter).OwningMenu = owningMenu;
+
+				MenuFlyout parentMenuFlyout;
+				parentMenuFlyout = parentMenuFlyoutPresenter.GetParentMenuFlyout();
+
+				if (parentMenuFlyout != null)
+				{
+					// Update the TemplateSettings before it is opened.
+					(m_tpPresenter as MenuFlyoutPresenter).SetParentMenuFlyout(parentMenuFlyout);
+					(m_tpPresenter as MenuFlyoutPresenter).UpdateTemplateSettings();
+
+					// Forward the parent presenter's properties to the sub presenter
+					ForwardPresenterProperties(
+						parentMenuFlyout,
+						parentMenuFlyoutPresenter,
+						m_tpPresenter as MenuFlyoutPresenter);
+				}
+			}
+
+			m_tpPopup.HorizontalOffset = position.X;
+			m_tpPopup.VerticalOffset = position.Y;
+			SetIsOpen(true);
+
+
+		}
+
+		void ISubMenuOwner.PositionSubMenu(Point position)
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: PositionSubMenu - (%f, %f).", this, position.X, position.Y));
+#endif // MFSI_DEBUG
+
+			if (position.X != float.NegativeInfinity)
+			{
+				m_tpPopup.HorizontalOffset = position.X;
+			}
+
+			if (position.Y != float.NegativeInfinity)
+			{
+				m_tpPopup.VerticalOffset = position.Y;
+			}
+
+
+		}
+
+		void ISubMenuOwner.ClosePeerSubMenus()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: ClosePeerSubMenus.", this));
+#endif // MFSI_DEBUG
+
+			EnsureCloseExistingSubItems();
+
+		}
+
+		void ISubMenuOwner.CloseSubMenu()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: CloseSubMenu.", this));
+#endif // MFSI_DEBUG
+
+			SetIsOpen(false);
+
+		}
+
+		void ISubMenuOwner.CloseSubMenuTree()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: CloseSubMenuTree.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.CloseChildSubMenus();
+
+		}
+
+		void ISubMenuOwner.DelayCloseSubMenu()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: DelayCloseSubMenu.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.DelayCloseSubMenu();
+		}
+
+		void ISubMenuOwner.CancelCloseSubMenu()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: CancelCloseSubMenu.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.CancelCloseSubMenu();
+		}
+
+		void ISubMenuOwner.RaiseAutomationPeerExpandCollapse(bool isOpen)
+		{
+			// UNO TODO
+			//AutomationPeer spAutomationPeer;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)